lua/starfall/toolscreen.lua

```diff
@@ -71,42 +130,33 @@ hook.Add("PreRender", "StarfallToolscreenPrepare", function()
 	hook.Remove("PreRender", "StarfallToolscreenPrepare")
 end)
 
-local stars_errored_delay = 0
 hook.Add("StarfallError", "StarfallToolscreenError", function(_, owner, client)
 	local local_player = LocalPlayer()
 	if owner ~= local_player then return end
 	if client and client ~= local_player then return end
-
-	if stars_errored_delay <= 0 then
-		for _, star in ipairs(stars) do
-			star.color = HSVToColor(350 + math_random() * 20, 0.7, 0.85)
-		end
-	end
-	stars_errored_delay = 3 -- How long should the stars remain red after an error
+	last_error_time = RealTime() + star_error_color_time
 end)
 
 -------------------------------------------
 
-local last_frame = 0
 local is_linux = system.IsLinux()
 local ply_eye_ang_prev = Angle()
 function SF.DrawToolgunScreen(w, h, title, scroll_text)
-	local curtime = RealTime()
-	local dt = curtime - last_frame
+	curtime = RealTime()
+	dt = curtime - last_frame
 	if dt > 1 / simulation_fps then
 		last_frame = curtime
-		stars_errored_delay = stars_errored_delay - dt
-
-		local deceleration = star_deceleration ^ dt
 		local ply = LocalPlayer()
 
 		local ply_eye_ang = ply:EyeAngles()
 		local ply_eye_ang_delta = ply_eye_ang - ply_eye_ang_prev
-		local ply_eye_pitch = math_normalize_angle(ply_eye_ang_delta[1]) * star_ply_angle_mul
-		local ply_eye_yaw = math_normalize_angle(ply_eye_ang_delta[2]) * star_ply_angle_mul
+
+		ply_eye_pitch = math_normalize_angle(ply_eye_ang_delta[1]) * star_ply_angle_mul
+		ply_eye_yaw = math_normalize_angle(ply_eye_ang_delta[2]) * star_ply_angle_mul
 		ply_eye_ang_prev = ply_eye_ang
+		deceleration = star_deceleration ^ dt
```

Comment:
I still think this pow with dt is weird.

Reply:
The formula seems to work perfectly, take a look at this simple Starfall script that simulates deceleration over X frames.
```lua
local function sim(vel, decel, fps)
    print(string.format("Simulating deceleration over %d frames", fps))
    local dt = 1 / fps
    for i = 1, fps do
        vel = vel * decel ^ dt
        print(math.truncate(vel, 2))
    end
end

for _, fps in ipairs({ 5, 10, 30 }) do
    sim(100, 0.1, fps)
end
```
```c
Simulating deceleration over 5 frames
63.09
39.81
25.11
15.84
10
Simulating deceleration over 10 frames
79.43
63.09
50.11
39.81
31.62
25.11
19.95
15.84
12.58
9.99
Simulating deceleration over 30 frames
92.61
85.76
79.43
73.56
68.12
63.09
58.43
54.11
50.11
46.41
42.98
39.81
36.86
34.14
31.62
29.28
27.12
25.11
23.26
21.54
19.95
18.47
17.11
15.84
14.67
13.59
12.58
11.65
10.79
9.99

```
All end up at `~10`, which is correct since velocity is a function of time `unit/time`, so the amount of FPS shouldn't affect it.  
  
That being said, I suck at maths, so if you have any better ideas, go at it :P
PS. This is where I got my answer from: https://stackoverflow.com/a/14014343.